src/Components/CreateImageWizard/formComponents/Oscap.tsx

```diff
@@ -17,37 +19,47 @@ import {
   SelectVariant,
 } from '@patternfly/react-core/deprecated';
 import { HelpIcon } from '@patternfly/react-icons';
-import PropTypes from 'prop-types';
 
 import OscapProfileInformation from './OscapProfileInformation';
 
 import {
+  DistributionProfileItem,
   useGetOscapCustomizationsQuery,
   useGetOscapProfilesQuery,
 } from '../../../store/imageBuilderApi';
 import { reinitFileSystemConfiguratioStep } from '../steps/fileSystemConfiguration';
 import { reinitPackagesStep } from '../steps/packages';
 
+// eslint-disable-next-line
+type ChangeType = <F extends keyof Record<string, any>>(
+  name: F,
+  // eslint-disable-next-line
+  value?: Record<string, any>[F]
+) => void;
+
 /**
  * Every time there is change on this form step's state, reinitialise the steps
  * that are depending on it. This will ensure that if the user goes back and
  * change their mind, going forward again leaves them in a coherent and workable
  * form state.
  */
-const reinitDependingSteps = (change) => {
+const reinitDependingSteps = (change: ChangeType) => {
   reinitFileSystemConfiguratioStep(change);
   reinitPackagesStep(change);
 };
-
+type ProfileSelectorProps = {
+  input: { name: string };
+};
 /**
  * Component for the user to select the profile to apply to their image.
  * The selected profile will be stored in the `oscap-profile` form state variable.
  * The Component is shown or not depending on the ShowSelector variable.
  */
-const ProfileSelector = ({ input }) => {
+
+const ProfileSelector = ({ input }: ProfileSelectorProps) => {
   const { change, getState } = useFormApi();
+  const [profileName, setProfileName] = useState<string | undefined>('');
   const [profile, setProfile] = useState(getState()?.values?.['oscap-profile']);
```

Comment:
Could be   `const [profile, setProfile] = useState(getState()?.values?.['oscap-profile'] ? getState()?.values?.['oscap-profile'] : []);` instead